shopelectro/selenium/elements/product.py

```diff
@@ -26,17 +31,43 @@ def remove_from_cart(self):
 
 class CatalogCard(Product):
 
-    def __init__(self, driver: SiteDriver, card_index: int):
+    def __init__(
+        self,
+        driver: SiteDriver,
+        *,
+        index: int = None,
+        id_: int = None,
+    ):
         """
         Ctor.
 
-        :param int card_index: The index number of the product card at a category page
+        :param int index: The index number of the product card at a category page
         """
         self.driver = driver
-        self.xpath = f'//*[@id="products-wrapper"]/div[{card_index}]/div[2]/div[5]/'
+
+        if (index is None and not id_) or (index and id_):
+            raise ValueError('Provide either index or id_ to work with card.')
+        self._id = id_
+        self._index = index
+
+    def _build_xpath(self, path=''):
+        product_xpath = '//*[@id="products-wrapper"]'
+
+        if self._id:
+            return f'{product_xpath}//*[@data-product-id="{self._id}"]/{path}'
+
+        # xpath indexes starts from 1
+        return f'{product_xpath}/div[{self._index + 1}]/{path}'
+
+    def vendor_code(self):
+        if self._id:
```

Comment:
I'll remove this predicate, because `self._id != vendor_code`. I made this because I forget that we are still using Product.id in front